contracts/child/ChildToken/ChildMintableERC721.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.6.6;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IChildToken} from "./IChildToken.sol";
+import {NetworkAgnostic} from "../../common/NetworkAgnostic.sol";
+import {ChainConstants} from "../../ChainConstants.sol";
+
+
+contract ChildMintableERC721 is ERC721, IChildToken, AccessControl, NetworkAgnostic, ChainConstants {
+    bytes32 public constant DEPOSITOR_ROLE = keccak256("DEPOSITOR_ROLE");
+    mapping (uint256 => bool) public withdrawnTokens;
+
+    constructor(
+        string memory name_,
+        string memory symbol_
+    ) public ERC721(name_, symbol_) NetworkAgnostic(name_, ERC712_VERSION, ROOT_CHAIN_ID) {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _setupRole(DEPOSITOR_ROLE, _msgSender());
+    }
+
+    modifier only(bytes32 role) {
+        require(hasRole(role, _msgSender()), "ChildMintableERC721: INSUFFICIENT_PERMISSIONS");
+        _;
+    }
+
+    function _msgSender()
```

Comment:
Should we use a common contract and extend with this method since multiple contracts are using it?

Reply:
Same with `only`

Reply:
Faced inheritance issues for `_msgSender()`, trying for `only`